File: Demo/GNNoticeViewTool/ViewController.m
//
//  ViewController.m
//  GNNoticeViewTool
//
//  Created by mac on 2018/3/27.
//  Copyright © 2018年 mac. All rights reserved.
//

#import "ViewController.h"
#import "GNNotice.h"

@interface ViewController ()

@end

@implementation ViewController

- (IBAction)showNoNet:(id)sender {
    [self.view showNoNetNoticeViewWithCenter:self action:@selector(showLoading:)];
}
- (IBAction)showNoData:(id)sender {
    [self.view showNodataNoticeView];
}
- (IBAction)hideNoticeView:(id)sender {
    [self.view hideNoticeView];
}
- (IBAction)showLoading:(id)sender {
    [self.view showLoadingNoticeView];
}


@end

File: swift/Demo/GNNoticeSwift/ViewController.swift
//
//  ViewController.swift
//  GNNoticeSwift
//
//  Created by naton_iOS on 2023/11/29.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

    }

    @IBAction func showLoading(_ sender: Any) {
        self.view.gn.showLoadingView()
    }
    
    @IBAction func showNoData(_ sender: Any) {
        self.view.gn.showNoDataView()
    }
    
    @IBAction func showLoadFailed(_ sender: Any) {
//        self.view.gn.showNoNetView(self, action: #selector(self.showLoading), for: UIControl.Event.touchUpInside, centerOffset: CGPointMake(0, 100))
        self.view.gn.showNoNetView(self, action: #selector(self.showLoading), for: UIControl.Event.touchUpInside, centerOffset: CGPointMake(0, 0))
    }
    @IBAction func hidePlaceHolderView(_ sender: Any) {
        self.view.gn.hidePlaceHolderView()
    }
}

